src/components/accessories/patientOPD/patientOPD.tsx

```diff
@@ -89,17 +90,34 @@ const PatientOPD: FunctionComponent = () => {
     }
   }, [dispatch, activityTransitionState]);
 
-  const onSubmit = (opdValues: OpdWithOperationRows) => {
+  const onSubmit = (opdValues: OpdWithOperatioRowDTO) => {
     setShouldResetForm(false);
-    opdValues.opd.patientCode = patient?.code;
-    opdValues.opd.age = patient?.age;
-    opdValues.opd.sex = patient?.sex;
-    opdValues.opd.userID = userId;
-    const opdToSave = { ...opdToEdit, ...opdValues.opd };
-    if (!creationMode && opdToEdit.code) {
-      dispatch(updateOpd(opdToEdit.code, opdToSave, opdValues.operationRows));
+    opdValues.opdDTO.patientCode = patient?.code;
+    opdValues.opdDTO.age = patient?.age;
+    opdValues.opdDTO.sex = patient?.sex;
+    opdValues.opdDTO.userID = userId;
+    opdValues.opdDTO.patientName =
+      patient?.firstName + " " + patient?.secondName;
+    const opdToSave = { ...opdToEdit.opdDTO, ...opdValues.opdDTO };
+    if (!creationMode && opdToEdit.opdDTO.code) {
+      dispatch(
+        (opdToEdit.opdDTO.code,
+        {
+          opdDTO: opdToSave,
+          operationRows: opdValues.operationRows,
+        } as OpdWithOperatioRowDTO)
+      );
     } else
-      dispatch(createOpd({ ...opdToSave, code: 0 }, opdValues.operationRows));
+      console.log("Non-techincal user", {
+        opdDTO: opdToSave,
+        operationRows: opdValues.operationRows,
+      });
+    dispatch(
+      createOpdWithOperationsRowsList({
+        opdDTO: { ...opdToSave, code: 0 },
+        operationRows: opdValues.operationRows,
+      })
```

Comment:
Could you please add the else's brackets and indent the code?

Reply:
Ok